### DISCUSSION TITLE
Snapshot triggered using repository dispatch

### DISCUSSION THREAD

**Discussion by ambravo** [engswee/flashpipe#38] [Show and tell]
Hi Eng Swee! 

I wanted to share a use case I implemented that might be worth including in the documentation:

I used the `repository_dispatch` event (documented [here](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#repository_dispatch)) to trigger a GitHub Action that takes a snapshot of the tenant state after promoting an artifact with SAP’s Transport Management Service. 

The workflow is triggered by the SAP Cloud Transport Management Service using the Alert Notification Service. Specifically, the event `TmsImportFinished` (Import Finished) (documented [here](https://help.sap.com/docs/alert-notification/sap-alert-notification-for-sap-btp/sap-cloud-transport-management-import-finished?locale=en-US)) calls the GitHub endpoint to trigger the `repository_dispatch` event.

Really useful for maintaining a branch in the repository that reflects each change introduced by transports in a specific tenant. It simplifies tracking changes and provides clear visibility into the state of the system after every transport.

![Overview - GithubAction TM-AN Alert Notification ARAUCO CF SCP CF ARAUCO - US3 - CPA - SAP BTP Cockpit - Vivaldi 2024-12-12 at 10 31 40 pm jpg](https://github.com/user-attachments/assets/f0ce3dd1-89cf-47ad-b244-7df79fc6fec9)

Bests,

Ariel

**Comment by engswee** [comment]
Would you be able to create a Markdown file in the docs folder for such a use case and create a PR for it? Thanks.
